vtex/omnishipping
# Fixed deactivating Lean Shipping

#### What is the purpose of this pull request?
### Fixed

- Checking if logisticsInfo have the same slas;
- Not deactivating `leanShipping` if there is only one item in cart;
- Selecting `giftRegistry` addresses on second purchase.

#### What problem is this solving?
While checking if `logisticsInfo` has the same delivery SLAs, isEqual was wrongly used, and only gave the wrong result when more than 3 items in the SLAs array. So I tried a different approach and updated unit tests.

Also `giftRegistry` addresses were not being selected due to a previous update in which it only considered `residential` addresses.

#### How should this be manually tested?

**Test case 1**
1. Add [items to cart](https://fernando--vtexgame1.myvtex.com/checkout/cart/add?&workspace=fernando&sku=31&qty=1&seller=1&sc=1)
2. Go to Shipping and type postal code `22071060`
3. It should successfully deactivate leanShipping

**Test case 2**
1. Add [items to cart](https://fernando--vtexgame1.myvtex.com/checkout/cart/add?&workspace=fernando&sku=31&qty=1&seller=1&sc=1&sku=283&qty=1&seller=1&sc=1)
2. Go to Shipping and type postal code `22071060`
3. It should successfully deactivate leanShipping

**Test case 3**
1. Add [items to cart](https://fernando--vtexgame1.myvtex.com/checkout/cart/add?&workspace=fernando&sku=31&qty=1&seller=1&sc=1&gr=21)
2. Identify as `second-purchase@mailinator.com`
3. `giftRegistry` address should be successfully selected

#### Screenshots or example usage
n/a

#### Types of changes
